p2p/chain.py

```diff
@@ -257,6 +272,8 @@ def _request_block_parts(
             self,
             headers: List[BlockHeader],
             request_func: Callable[[ETHPeer, List[BlockHeader]], None]) -> int:
+        if not self.peer_pool.peers:
+            raise NoConnectedPeers()
         length = math.ceil(len(headers) / len(self.peer_pool.peers))
```

Comment:
Do you think changing this to use a `try/except` with `ZeroDivisionError` so that we are guaranteed to not have a race condition here would be good (peer pool empties between the check on 275 and this line).

Reply:
That can't happen because nothing yields control between line 275 and here

Reply:
I was more advocating that approach as a future proof approach since it's possible that future refactors change the structure in some minor way that it becomes possible.  `try/except` pattern provides a stronger link.  Not a big deal.  I may throw it in here in some of my other cleanup. 👍

Reply:
Oh, sorry, I misunderstood you. I thought you meant to catch `ZeroDivisionError` in addition to checking, but now I think you suggested I do that instead of the check? I will include that change in another PR, unless you beat me to it